components/builder-datastore/hooks/install

```diff
@@ -0,0 +1,506 @@
+    --host={{cfg.datastore.host}} \
+    --port={{cfg.datastore.port}} \
+    --username={{cfg.datastore.user}} \
+    --dbname={{cfg.datastore.database}} \
+    --no-password \
+    --quote-all-identifiers \
+    --file="$pg_dump_file"
+
+  # shellcheck disable=SC2181
+  if [[ $? == 0 ]]; then
+    echo "pg_dump was successful"
+  else
+    echo "pg_dump FAILED"
+    exit 16
+  fi
+}
+
+load_postgres_database() {
+
+  echo Loading the builder database from file generated via pg_dump using psql
+
+  sudo -E hab pkg exec "$ACTIVE_IDENT" psql -- \
+    --host={{cfg.datastore.host}} \
+    --port={{cfg.datastore.port}} \
+    --username={{cfg.datastore.user}} \
+    --dbname={{cfg.datastore.database}} \
+    --no-password \
+    --file="$pg_dump_file"
+
+  local -ri exit_status=$?
+  if DEBUG; then
+    local what_happened=""
+    case $exit_status in
+    0)
+      what_happened="success"
+      ;;
+    1)
+      what_happened="fatal psql error"
+      ;;
+    2)
+      what_happened="bad server connection, session not interactive"
+      ;;
+    3)
+      what_happened="sql script error"
+      ;;
+    *)
+      what_happened="unexpected exit status"
+      ;;
+    esac
+    printf "%s: exit status: %s (%s)\n" "${FUNCNAME[0]}" $exit_status "$what_happened"
+  fi
+  if [[ $exit_status -eq 0 ]]; then
+    echo "Succeeded in loading the builder database from the pg_dump file"
+  else
+    echo "Failed to load the pg_dump file"
+    exit 17
+  fi
+  return $exit_status
+}
+
+upgrade_postgres() {
+
+  echo Upgrading PostgreSQL
+
+  local -r work_dir="{{pkg.svc_var_path}}/tmp/upgrades/$MOMENT"
+  local -r pg_dump_file="$work_dir/pg_dump.sql"
+  local -r previous_dir="$work_dir/previous"
+  mkdir -p "$work_dir"
+
+  PGPASSWORD="$(cat "$PWFILE")"
+  export PGPASSWORD
+
+  manage_upgrade_readme "$work_dir"
+
+  ACTIVE_IDENT=$INSTALLED_PG_IDENT
+
+  start_postgres
+
+  # https://www.postgresql.org/docs/17/upgrading.html
+  # STEP 1) Making a backup, making sure that your database is not being updated
+  dump_postgres_database
+
+  # STEP 2) Shut down the old server
+  stop_postgres
+
+  # STEP 3) Rename the old installation directory
+  sudo mv "{{pkg.svc_data_path}}" "$previous_dir"
+
+  # STEP 4) Install the new version of PostgreSQL (as outlined in Chapter 17)
+  # This is a mostly a noop for us as this is accomplished as part of the hab
+  # svc but we do need to manage the active_ident
+  ACTIVE_IDENT=$PACKAGED_DATASTORE_IDENT
+
+  # STEP 5) If needed, create a new database cluster. Remember you must be logged in as the special database user account
+  install_postgres
+
+  # STEP 6) Restore your previous pg_hba.conf and any postgresql.conf modifications
+
+  # STEP 7) Start the database server (again using the special database user account)
+  start_postgres
+
+  # step 8) finally, restore your data from backup
+  load_postgres_database
+  stop_postgres # because the run method is going to start it again
+  sudo hab svc load "$ACTIVE_IDENT" --force
```

Comment:
we should already be loaded.

Reply:
I'm pretty sure that I actually need that to ensure that the templates are recompiled and written out to the config directory. I felt like I shouldn't need it but I tried a variety of things as I worked on this and a this hab svc load seemed to be the minimal and best take.

Reply:
Yes, the three conf files in /home/admin/Development/progress/builder/components/builder-datastore/config--pg_hba.conf, pg_ident.conf, and postgresql.conf--need to be updated.  Without this call I don't think they get recompiled.  I think we get to skate by with the first two but there are changes in postgresql.conf that are necessary.

Reply:
After discussing will comment out and retest to verify need.

Reply:
Removed